lib/shopify_api/resources/smart_collection.rb

```diff
@@ -4,10 +4,10 @@ class SmartCollection < Base
     include Metafields
 
     def products(options = {})
-      if options.key?(:only_sorted)
-        Product.find(:all, from: "/admin/smart_collections/#{id}/products.json", params: options)
-      else
+      if options.empty?
         Product.find(:all, params: { collection_id: id })
+      else
+        Product.find(:all, from: "/admin/smart_collections/#{id}/products.json", params: options)
       end
```

Comment:
We already test this on https://github.com/Shopify/shopify_api/blob/master/test/smart_collection_test.rb